integrations/docs/API/swagger.yaml

```diff
@@ -0,0 +1,287 @@
+openapi: 3.0.0
+info:
+  description: This is the Integration & Catalog API
+  version: "1.0.0"
+  title: Integration API
+  license:
+    name: Apache 2.0
+    url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
+tags:
+  - name: admins
+    description: Secured Admin-only calls
+  - name: developers
+    description: Operations available to regular developers
+paths:
+  /catalog:
+    get:
+      tags:
+        - developers
+      summary: schema list of loaded catalogs
+      operationId: searchCatalog
+      description: |
+        By passing in the appropriate search attributes, you can search for
+        available catalogs in the system
+      parameters:
+        - in: query
+          name: searchString
+          description: pass an optional search string for looking up catalog components
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: limit
+          description: maximum number of records to return
+          schema:
+            type: integer
+            format: int32
+            minimum: 0
+            maximum: 50
+      responses:
+        '200':
+          description: search results matching criteria
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '../../../schema/system/catalog.schema'
+        '400':
+          description: bad input parameter
+    post:
+      tags:
+        - admins
+      summary: adds a catalog item
+      operationId: addCatalog
+      description: |
+        Adds a catalog item to the system, expecting the internal URLs to be accessible
+      requestBody:
+        description: Inventory item to add
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '../../../schema/system/catalog.schema'
+      responses:
+        '201':
+          description: item created
+        '400':
+          description: 'invalid input, object invalid'
+        '409':
+          description: an existing item already exists
+  /repository:
+    get:
+      tags:
+        - developers
+      summary: repository of available integrations
+      operationId: searchRepository
+      description: |
+        By passing in the appropriate search attributes, you can search for
+        available integration in the repository (available for loading into the store)
+      parameters:
+        - in: query
+          name: searchString
+          description: pass an optional search string for looking up available Integrations
+          required: false
+          schema:
+            type: string
```

Comment:
For this one, I do not see a separate entry in this swagger for /repository/${id}. I think it makes sense to separate it out - for /repository?filter we can return all integrations, but only necessary to show on UI - like name, description, img url, etc.
Then for /repository/${id} we will need to fetch things like SSFO fields, etc.

Reply:
I was also checking the schema/sample json files - I couldn't find one with SSFO fields listed - only relative path to some schema file. / @YANG-DB can you point out where that would be/ add it if not present? Front end would need to get the actual mapping/fields, or is the plan to expose another API to query the fields?